pyvelox/pyvelox.h

```diff
@@ -19,33 +19,211 @@
+template <typename T>
+inline VectorPtr variantsToFlatVector(
+    const TypePtr& type,
+    const std::vector<velox::variant>& variants,
+    facebook::velox::memory::MemoryPool* pool) {
+  std::shared_ptr<FlatVector<T>> result =
+      BaseVector::create<FlatVector<T>>(type, variants.size(), pool);
+  for (int i = 0; i < variants.size(); i++) {
+    if (variants[i].isNull()) {
+      result->setNull(i, true);
+    } else {
+      result->set(i, variants[i].value<T>());
+    }
+  }
+  return result;
+}
+
+template <>
+inline VectorPtr variantsToFlatVector<velox::StringView>(
+    const TypePtr& type,
+    const std::vector<velox::variant>& variants,
+    facebook::velox::memory::MemoryPool* pool) {
+  velox::StringViewBufferHolder holder(pool);
+  std::shared_ptr<FlatVector<velox::StringView>> result =
+      BaseVector::create<FlatVector<velox::StringView>>(
+          type, variants.size(), pool);
+  for (int i = 0; i < variants.size(); i++) {
+    if (variants[i].isNull()) {
+      result->setNull(i, true);
+    } else {
+      velox::StringView view =
+          holder.getOwnedValue(variants[i].value<std::string>());
+      result->set(i, view);
+    }
+  }
+  result->setStringBuffers(holder.moveBuffers());
+  return result;
+}
+
+inline VectorPtr pyListToVector(
+    const pybind11::list& list,
+    facebook::velox::memory::MemoryPool* pool) {
+  namespace py = pybind11;
+  std::vector<velox::variant> variants;
+  variants.reserve(list.size());
+  for (auto item : list) {
+    variants.push_back(pyToVariant(item));
+  }
+
+  if (variants.empty()) {
+    throw py::value_error("Can't create a Velox vector from an empty list");
+  }
+
+  velox::TypeKind first_kind = velox::TypeKind::INVALID;
+  for (velox::variant& var : variants) {
+    if (var.hasValue()) {
+      if (first_kind == velox::TypeKind::INVALID) {
+        first_kind = var.kind();
+      } else if (var.kind() != first_kind) {
+        throw py::type_error(
+            "Velox Vector must consist of items of the same type");
+      }
+    }
+  }
+
+  if (first_kind == velox::TypeKind::INVALID) {
+    throw py::value_error(
+        "Can't create a Velox vector consisting of only None");
+  }
+
+  TypePtr type = fromKindToScalerType(first_kind);
+  switch (first_kind) {
+    case velox::TypeKind::BOOLEAN:
+      return variantsToFlatVector<bool>(type, variants, pool);
+    case velox::TypeKind::BIGINT:
+      return variantsToFlatVector<int64_t>(type, variants, pool);
+    case velox::TypeKind::DOUBLE:
+      return variantsToFlatVector<double>(type, variants, pool);
+    case velox::TypeKind::VARCHAR:
+      return variantsToFlatVector<velox::StringView>(type, variants, pool);
+    default:
+      throw std::runtime_error("Impossible type detected for velox::variant");
+  }
+}
+
+template <typename T>
+inline pybind11::object getItemFromVector(
+    std::shared_ptr<BaseVector>& v,
+    int64_t idx) {
+  namespace py = pybind11;
+  const FlatVector<T>* flat = v->asFlatVector<T>();
```

Comment:
nit: in these cases feel free to use auto since its apparent what type of flat is.